sensu/resource_filter.go

```diff
@@ -56,7 +56,13 @@ func resourceFilterCreate(d *schema.ResourceData, meta interface{}) error {
 
 	expressions := expandStringList(d.Get("expressions").([]interface{}))
 	runtimeAssets := expandStringList(d.Get("runtime_assets").([]interface{}))
-	when := expandTimeWindows(d.Get("when").(*schema.Set).List())
+	// to prevent the creation of an empty when block
+	whenRaw, whenExist := d.GetOk("when")
+	expandedTimeWindows := expandTimeWindows(whenRaw.(*schema.Set).List())
+	when := &expandedTimeWindows
+	if !whenExist {
```

Comment:
would it maybe be better to initialize `when = nil` and then set it to `expandedTimeWindows := expandTimeWindows(whenRaw.(*schema.Set).List())` if `whenExists`?